stdlib/winreg.pyi

```diff
@@ -24,9 +25,7 @@ if sys.platform == "win32":
     def QueryValueEx(key: _KeyType, name: str, /) -> tuple[Any, int]: ...
     def SaveKey(key: _KeyType, file_name: str, /) -> None: ...
     def SetValue(key: _KeyType, sub_key: str, type: int, value: str, /) -> None: ...
-    def SetValueEx(
-        key: _KeyType, value_name: str | None, reserved: Any, type: int, value: str | int, /
-    ) -> None: ...  # reserved is ignored
+    def SetValueEx(key: _KeyType, value_name: str | None, reserved: Unused, type: int, value: str | bytes | int, /) -> None: ...
```

Comment:
I could also add an overload specific for `REG_BINARY`, but I feel like at that point I may as well complete the overloads for all possible types.

```suggestion
    @overload  # type=REG_BINARY
    def SetValueEx(key: _KeyType, value_name: str | None, reserved: Unused, type: Literal[3], value: bytes, /) -> None: ...
    @overload
    def SetValueEx(key: _KeyType, value_name: str | None, reserved: Unused, type: int, value: str | bytes | int, /) -> None: ...
```

Reply:
Looking at the C code in CPython (function `Py2Reg` in `winreg.c`), the allowed types are:
- `REG_DWORD`: `int | None`
- `REG_QWORD`: `int | None`
- `REG_SZ`, `REG_EXPAND_SZ`: `str | None`
- `REG_MULTI_SZ`: `None | list[str]`
- `REG_BINARY` and everything else: `None | ReadableBuffer`

That sums up to:
```suggestion
    def SetValueEx(key: _KeyType, value_name: str | None, reserved: Unused, type: int, value: str | ReadableBuffer | list[str] | int | None, /) -> None: ...
```

But we could also add overloads for the individual types; I don't feel strongly.

Reply:
Thanks for looking into the implementation for this one, the method was missing more than I was expecting.

Since you already did the work of mapping value to their type here, I may as well add the overloads.